mux.go

```diff
@@ -13,8 +13,11 @@ import (
 )
 
 var (
+	// ErrMethodMismatch is returned when the error in the request does not match
```

Comment:
I suspect "when the **error** in the request" should be "when the **method** in the request".